### PR TITLE
Allow explicit dimensions to override defaults

### PR DESCRIPTION
**Pull request by ramn**

### CODE REVIEW

**Comment by ramn** [comment]
@Marwes

File: src/collector.rs [hunk 1]
```diff
@@ -294,9 +294,15 @@ impl Collector {
             name: l.key().to_owned(),
             value: l.value().to_owned(),
         });
-        self.default_dimensions()
+        let all_dims: BTreeMap<_, _> = self
+            .default_dimensions()
             .chain(dimensions_from_keys)
+            .map(|d| (d.name.clone(), d))
+            .collect();
```
Comment (Marwes) [inline src/collector.rs:306]: Would be faster to collect into a `Vec`, (stable) sort and dedup it by dimension name https://doc.rust-lang.org/std/vec/struct.Vec.html#method.dedup .

**Review by Marwes — CHANGES_REQUESTED** [review]

File: src/collector.rs (hunk 1 above)
Comment (ramn) [reply src/collector.rs:306]: I tried that, but it was tricky to get {sort,dedup}_by_key to work with references, so it became way more ugly. I think I hit this one https://github.com/rust-lang/rust/issues/34162

Comment (Marwes) [reply src/collector.rs:306]: `sort_by`/`dedup_by` should work in that case.

Comment (Marwes) [reply src/collector.rs:306]: Nit: This only need to be sorted and deduped if there are extra dimensions passed

**Pull request closed without merging**